test/IRGen/ELF-remove-autolink-section.swift

```diff
@@ -0,0 +1,11 @@
+// RUN: %swiftc_driver -target x86_64-unknown-linux-gnu -emit-ir %s  | %FileCheck %s -check-prefix ELF
+// RUN: %swiftc_driver -c -target x86_64-unknown-linux-gnu  %s -o %t1.obj
+// RUN: %swiftc_driver    -target x86_64-unknown-linux-gnu  %s -o %t2.exe
+// RUN: llvm-readelf -S %t1.obj | %FileCheck %s -check-prefix SECTION
+// RUN: llvm-readelf -S %t2.exe | %FileCheck %s -check-prefix NOSECTION
+
```

Comment:
I think that you need to add a `// REQUIRES: OS=linux` here since the standard library will not be available otherwise.  At that point you should be able to simplify the rules to something like:

```
// RUN: %swiftc -emit-ir %s -o - | %FileCheck %s -check-prefix ELF
// RUN: %swiftc -emit-executable -o %t %s
// RUN: %swiftc -c %s -o - | llvm-readelf -S | %FileCheck %s -check-prefix SECTION
// RUN: llvm-readelf -S %t | %FileCheck %s -check-prefix NOSECTION
```